manifests/candlepin.pp

```diff
@@ -76,6 +75,8 @@
   $alias = 'candlepin-ca'
 
   if $deploy {
+    $artemis_client_dn = certs::certificate_subject($client_cert)
```

Comment:
I still suspect we may need to use Deferred. Regular functions run on the Puppetserver while the certificate files are on the agent. While it may work in our installer where everything is on the same machine, I think it'll break in regular operation.